polystores/stores/s3_store.py

```diff
@@ -204,7 +204,17 @@ def get_bucket(self, bucket_name):
         return self.resource.Bucket(bucket_name)
 
     def ls(self, path):
-        results = self.list(bucket_name=path)
+        # Strip out any unnecessary characters
+        stripped = path.replace('s3://', '').rstrip('/')
```

Comment:
I have the feeling that this logic should not be here, I also think that this change will fix this command, but probably other commands will break because they will require similar logic?

Reply:
is it possible to reuse the logic from [parse_s3_url](https://github.com/polyaxon/polystores/pull/13/files#diff-dc7f4457be183c2d31519b26fbaab030R162)

Reply:
Ah, I didn't see that function. It'll probably works so I'll try it out. I don't think other commands will break due to this one being fixed. Since few things are using the actual `ls` command and it can handle a few cases just fine.

Reply:
Using `parse_s3_url` works for all the cases other than ones that don't include the `s3://` such as https://github.com/polyaxon/polystores/pull/13/files/8aa63458e286144225b160e62cddaeb127d59651#diff-7eb1c404535b20b921b202eac500ff28R98 Is that ok? I wasn't exactly sure what all cases would be passed to this function, so I leaned on covering more and being more defensive with my solution. If you're ok with it though, I can switch over to using `parse_s3_url`

Reply:
yeah I don't think we would pass a path without a netloc, that validation ensures that the path implements a valid S3 protocol, not necessarly S3, it could be minio or any endpoint that implements the protocol. So I would say let;'s consolidate around the same logic used by all other methods.

Reply:
Done!